pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/TimeSegmentPruner.java

```diff
@@ -478,6 +486,21 @@ public synchronized void refreshSegment(String segment) {
     return res;
   }
 
+  private List<Interval> getComplementSortedIntervals(List<Interval> intervals) {
+    List<Interval> res = new ArrayList<>();
+    long startTime = MIN_START_TIME;
+    for (Interval interval : intervals) {
+      if (interval._min > startTime) {
```

Comment:
Is this logic correct? E.g. what if you have 3 intervals:
[2, 10] ,[5,15], [13, 20]
the intervals are [0, 1], [11, 12], [21, MAX_END_TIME], but should just be [0,1], [21, MAX_END_TIME] ?
Also you can skip the interval if `startTime = interval._min - 1`

Reply:
The intervals passed in has no overlapping (guaranteed by AND and OR handling). Will add some javadoc explaining that.
We cannot skip the interval because start does not include the previous `max`, e.g. `[1, 2], [4, 5]` -> `[0, 0], [3, 3], [6, MAX]`

Reply:
After some consideration, I think I would skip `NOT` handling for this class and address it in a separate PR with better testing coverage